docs/literate/src/files/hohqmesh_literate.jl

```diff
@@ -0,0 +1,520 @@
+# the `elixir` file used to run the simulation.
+
+# Finally, if you want to convert all the solution files to VTK execute
+# ```julia
+# trixi2vtk("out/solution_000*.h5", output_directory="out", nvisnodes=12)
+# ```
+# then it is possible to open the `.pvd` file with ParaView and create a video of the simulation.
+
+# ## Creating a mesh using HOHQMesh
+
+# The creation of an unstructured quadrilateral mesh using HOHQMesh.jl is driven by a **control file**. In this file the user dictates
+# the domain to be meshed, prescribes any desired boundary curvature, the polynomial order of said boundaries, etc.
+# In this tutorial we cover several basic features of the possible control inputs. For a complete discussion
+# on this topic see the [HOHQMesh control file documentation](https://trixi-framework.github.io/HOHQMesh/the-control-file/).
+
+# To begin, we provide a complete control file in this tutorial. After this we give a breakdown
+# of the control file components to explain the chosen parameters.
+
+# Suppose we want to create a mesh of a domain with straight sided
+# outer boundaries and a curvilinear "ice cream cone" shaped object at its center.
+
+# ![mesh_boundary_cartoon](https://user-images.githubusercontent.com/25242486/129603954-9788500d-bba8-49be-8e6f-7555099dbf7c.png)
+
+# The associated `ice_cream_straight_sides.control` file is given below.
+# ```
+# \begin{CONTROL_INPUT}
+#    \begin{RUN_PARAMETERS}
+#       mesh file name   = mesh_ice_cream_straight.mesh
+#       plot file name   = mesh_ice_cream_straight.tec
+#       stats file name  = none
+#       mesh file format = ISM-v2
+#       polynomial order = 4
+#       plot file format = skeleton
+#    \end{RUN_PARAMETERS}
+#
+#    \begin{BACKGROUND_GRID}
+#        x0 = [-8.0, -8.0, 0.0]
+#        dx = [1.0, 1.0, 0.0]
+#        N  = [16,16,1]
+#    \end{BACKGROUND_GRID}
+#
+#    \begin{SPRING_SMOOTHER}
+#       smoothing            = ON
+#       smoothing type       = LinearAndCrossBarSpring
+#       number of iterations = 25
+#    \end{SPRING_SMOOTHER}
+#
+# \end{CONTROL_INPUT}
+#
+# \begin{MODEL}
+#
+#    \begin{INNER_BOUNDARIES}
+#
+#       \begin{CHAIN}
+#          name = IceCreamCone
+#          \begin{END_POINTS_LINE}
+#             name = LeftSlant
+#             xStart = [-2.0, 1.0, 0.0]
+#             xEnd   = [ 0.0, -3.0, 0.0]
+#          \end{END_POINTS_LINE}
+# 
+#          \begin{END_POINTS_LINE}
+#             name = RightSlant
+#             xStart = [ 0.0, -3.0, 0.0]
+#             xEnd   = [ 2.0, 1.0, 0.0]
+#          \end{END_POINTS_LINE}
+# 
+#          \begin{CIRCULAR_ARC}
+#             name        = IceCream
+#             units       = degrees
+#             center      = [ 0.0, 1.0, 0.0]
+#             radius      = 2.0
+#             start angle = 0.0
+#             end angle   = 180.0
+#          \end{CIRCULAR_ARC}
+#       \end{CHAIN}
+#
+#    \end{INNER_BOUNDARIES}
+# 
+# \end{MODEL}
+# \end{FILE}
+# ```
```

Comment:
This is the complete control file content, isn't it? If so, you could avoid duplicating everything by using something like
```
julia> open(filename, "w") do io
           println(io, raw"""
       \begin{CONTROL_INPUT}
           \begin{RUN_PARAMETERS}
              mesh file name   = mesh_ice_cream_straight.mesh
              plot file name   = mesh_ice_cream_straight.tec
              stats file name  = none
              mesh file format = ISM-v2
              polynomial order = 4
              plot file format = skeleton
           \end{RUN_PARAMETERS}

           \begin{BACKGROUND_GRID}
               x0 = [-8.0, -8.0, 0.0]
               dx = [1.0, 1.0, 0.0]
               N  = [16,16,1]
           \end{BACKGROUND_GRID}

           \begin{SPRING_SMOOTHER}
              smoothing            = ON
              smoothing type       = LinearAndCrossBarSpring
              number of iterations = 25
           \end{SPRING_SMOOTHER}

        \end{CONTROL_INPUT}

        \begin{MODEL}

           \begin{INNER_BOUNDARIES}

              \begin{CHAIN}
                 name = IceCreamCone
                 \begin{END_POINTS_LINE}
                    name = LeftSlant
                    xStart = [-2.0, 1.0, 0.0]
                    xEnd   = [ 0.0, -3.0, 0.0]
                 \end{END_POINTS_LINE}
        
                 \begin{END_POINTS_LINE}
                    name = RightSlant
                    xStart = [ 0.0, -3.0, 0.0]
                    xEnd   = [ 2.0, 1.0, 0.0]
                 \end{END_POINTS_LINE}
        
                 \begin{CIRCULAR_ARC}
                    name        = IceCream
                    units       = degrees
                    center      = [ 0.0, 1.0, 0.0]
                    radius      = 2.0
                    start angle = 0.0
                    end angle   = 180.0
                 \end{CIRCULAR_ARC}
              \end{CHAIN}

           \end{INNER_BOUNDARIES}
        
        \end{MODEL}
        \end{FILE}
       """)
       end
```

Reply:
Yes it is. That's a good idea.

Reply:
I also added some code for the short code snippets
```
println(match(r"[^\S\n]*\\begin{RUN_PARAMETERS}[\s\S]+\\end{RUN_PARAMETERS}", read("out/ice_cream_straight_sides.control", String)).match)
```
So now, there's no duplicate code at all.
Do you think it's too complicated and confusing?

Reply:
Yes, I think so. Many people working with Trixi.jl will not necessarily be used to regex stuff. Since the text around that control file snippet is also some kind of code duplication (since it refers to the code explicitly), it's better to keep this simple for now, I think.